test/views/pages/map/dynamic_map_test.dart

```diff
@@ -0,0 +1,286 @@
+
+      // Run the tests twice, because the first time we click on a chip, we may
+      // not refresh the posts (the first chip may be the default value)
+      for (int i = 0; i < 2; ++i) {
+        for (final option in MapSelectionOptions.values) {
+          // Click on option chip
+          final optionChip = find.byKey(
+            MapSelectionOptionChips.optionChipKeys[option]!,
+          );
+          expect(optionChip, findsOneWidget);
+          await tester.tap(optionChip);
+          await tester.pumpAndSettle();
+
+          // Verify the option
+          await testOption(container, option);
+        }
+      }
+    });
+
+    /// Runs a test where we navigate to a different page, and then come back to the map, in order
+    /// to see if the pins were correctly refreshed. An example is that we delete a post, and then
+    /// come back to the map to see if the pin was removed.
+    /// The [testTitle] is the title of the test, [optionToTest] is the map pin option we want to
+    /// have selected during the test, [expectedPinDelta] is the expected difference in the number
+    /// of pins after the protocol is run, and [protocol] is the function that runs the protocol
+    /// (this is where one would delete a post, or create a new post, for instance).
+    void testNavigation({
+      required String testTitle,
+      required MapSelectionOptions optionToTest,
+      required int expectedPinDelta,
+      required Future<void> Function(
+        WidgetTester tester,
+        ProviderContainer container,
+      ) protocol,
+    }) {
+      testWidgets(testTitle, (tester) async {
+        final container = await beginTest(tester);
+
+        // Tap on the option chip we want for the test
+        final chip = find.byKey(
+          MapSelectionOptionChips.optionChipKeys[optionToTest]!,
+        );
+        expect(chip, findsOneWidget);
+        await tester.tap(chip);
+
+        // Verify we have the correct number of pins
+        final expectedPins =
+            expectedPostsForOption[optionToTest] ?? List.empty();
+        final pinsStart = await container.read(mapPinViewModelProvider.future);
+        expect(pinsStart, hasLength(expectedPins.length));
+
+        // Run the protocol
+        await protocol(tester, container);
+
+        // Verify the final state, the option should not have changed
+        final currentOption = container.read(
+          mapSelectionOptionsViewModelProvider,
+        );
+        expect(currentOption, equals(optionToTest));
+
+        // The pins should have the correct number
+        final pinsEnd = await container.read(mapPinViewModelProvider.future);
+        expect(pinsEnd, hasLength(expectedPins.length + expectedPinDelta));
+      });
+    }
+
+    testNavigation(
+      testTitle: "Pins refresh after post creation",
+      optionToTest: MapSelectionOptions.nearby,
+      expectedPinDelta: 1,
+      protocol: (tester, container) async {
+        await tester.tap(find.text("New post"));
```

Comment:
I think that this line can be removed.

```suggestion
```

Reply:
Oh, this is an excellent catch good job! This line should definitely not have found its way here, thank you so much for having found it!
Fixed in e9e5f52e206a6135ce9298589efdf5a05c8c5d49.